ceph/ceph_admin/dashboard.py

```diff
@@ -10,6 +11,29 @@
 LOG = Log(__name__)
 
 
+def execute_commands(cls, commands):
+    for cmd in commands:
+        out, err = cls.shell(args=[cmd])
+        LOG.info("STDOUT:\n %s" % out)
+        LOG.error("STDERR:\n %s" % err)
+
+
+def validate_url(url):
+    """Method to validate provided API URL.
+    The HTTP 200 OK success status response code indicates that the request has succeeded.
+
+    Args:
+        url: URL of any API.
+    """
+    url = url.rstrip()
+    resp = requests.get(url, verify=False)
+
+    if resp.status_code == 200:
+        LOG.info("API validation is successful")
+    else:
+        LOG.error(f"API validation fails with status code:{resp.status_code}")
```

Comment:
Shouldn't you raise an exception if there is verification check failure? Is this considered to be a soft error?

Reply:
when i tested at multiple times sometime due to network issues API validation checks got failed, If we raise an exception then the entire suite will fail bcz of some known issues Hence I thought it's good to keep it as soft error.